src/pgduckdb_planner.cpp

```diff
@@ -6,34 +6,107 @@
 #include "pgduckdb/scan/postgres_scan.hpp"
 #include "pgduckdb/pgduckdb_types.hpp"
 #include "pgduckdb/pgduckdb_planner.hpp"
+#include "pgduckdb/pgduckdb_table_am.hpp"
 
 extern "C" {
 #include "postgres.h"
+
 #include "access/xact.h"
+#include "access/table.h"
 #include "catalog/pg_type.h"
 #include "nodes/makefuncs.h"
 #include "nodes/nodes.h"
+#include "nodes/nodeFuncs.h"
 #include "nodes/params.h"
 #include "optimizer/optimizer.h"
 #include "optimizer/planner.h"
 #include "optimizer/planmain.h"
+#include "parser/parse_coerce.h"
+#include "rewrite/rewriteHandler.h"
 #include "tcop/pquery.h"
+#include "utils/typcache.h"
+#include "utils/rel.h"
 #include "utils/syscache.h"
 #include "utils/guc.h"
 
 #include "pgduckdb/pgduckdb_ruleutils.h"
 }
 
+#include "pgduckdb/pg/types.hpp"
 #include "pgduckdb/pgduckdb_duckdb.hpp"
 #include "pgduckdb/pgduckdb_node.hpp"
 #include "pgduckdb/vendor/pg_list.hpp"
 #include "pgduckdb/utility/cpp_wrapper.hpp"
 #include "pgduckdb/pgduckdb_types.hpp"
 
+bool
+IsAllowedPostgresInsert(Query *query, bool throw_error) {
+	int elevel = throw_error ? ERROR : DEBUG4;
+	if (query->commandType != CMD_INSERT) {
+		elog(elevel, "DuckDB only supports INSERT/SELECT on Postgres tables");
+		return false;
+	}
+
+	Assert(list_length(query->rtable) >= query->resultRelation);
+	RangeTblEntry *target_rel = (RangeTblEntry *)list_nth(query->rtable, query->resultRelation - 1);
+	if (pgduckdb::IsDuckdbTable(target_rel->relid)) {
+		return false;
+	}
+
+	/* Checking supported INSERT types */
+	RangeTblEntry *select_rte = NULL;
+	foreach_node(RangeTblEntry, rte, query->rtable) {
+		if (rte->rtekind == RTE_SUBQUERY) {
+			select_rte = rte;
+		}
+	}
+
+	if (!select_rte) {
+		elog(elevel, "DuckDB does not support INSERT without a subquery");
+		return false;
+	}
+	
+	/* TODO: Checking supported target column types */
+	Relation rel = RelationIdGetRelation(target_rel->relid);
+	TupleDesc target_desc = RelationGetDescr(rel);
+	bool ret = true;
+	for (int i = 0; i < target_desc->natts; i++) {
+		Form_pg_attribute attr = TupleDescAttr(target_desc, i);
+		if (attr->attisdropped) {
+			continue;
+		}
+
+		if (attr->atttypid == BYTEAOID || attr->atttypid == XMLOID || pgduckdb::pg::IsDomainType(attr->atttypid) ||
```

Comment:
Instead of hardcoding some unsupported types, I think we should instead determine the allowed types. Also, why are these types unsupported? Because we do support BYTEA as a type for regular duckdb execution, so that seems surprising that it doesn't work here.

Reply:
Changed the behavior to whitelist. As long as the Postgres type can be converted to a DuckDB type using `ConvertPostgresToDuckColumnType`, it will be allowed.